java/vector/src/main/java/org/apache/arrow/vector/validate/Status.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.validate;
+
+/**
+ * Status used for vector and visitor.
+ */
+public class Status {
```

Comment:
I think this might be to close of a copy from C++, I think we should probably throw an exception when a validation error occurs and have validate not return anything.  C++ has a Status class because it doesn't use exceptions (but this isn't idiomatic java).

If you want to clean this up I can do a more thorough review in the next few days.

Reply:
thanks for your comments, I have removed it.